client-spark/spark3/src/main/java/org/apache/spark/shuffle/writer/RssShuffleWriter.java

```diff
@@ -442,6 +442,8 @@ public Option<MapStatus> stop(boolean success) {
         return Option.empty();
       }
     } finally {
+      // cancel all async thread task related
+      sendingSet.stream().forEach(eventTask -> eventTask.cancel(true));
```

Comment:
Is `cancel` certain to succeed? If `cancel` returns false, what negative impact will it have?